File: code20260423/code/src/api/datacollection.ts
import axios from "axios";

const baseURL =
    import.meta.env.VITE_MACHINE_CONNECTION_API ?? "/machine-connection";

const client = axios.create({
    baseURL,
    timeout: 120_000,
    headers: { "Content-Type": "application/json" },
});

/**
 * 采集协议类型。
 * - IndustrialIoT：默认，走上游 IndustrialIoT.Host 的 /api/data/{id}/read。
 * - NCLinkApi：华中 NC-Link，走 nclink-api-server 的 /v1/{deviceId}/data/。
 */
export type DatacollectionProtocol = "IndustrialIoT" | "NCLinkApi";

export interface DatacollectionRow {
    id: number;
    name: string;
    path: string;
    datatype: string;
    collectionFrequency: number;
    /** ISO 日期字符串或 null */
    datetime?: string | null;
    protocol?: DatacollectionProtocol;
}

export interface DatacollectionSyncItem {
    name: string;
    path: string;
    datatype: string;
    collectionFrequency: number;
    /** 未传时后端按 IndustrialIoT 处理；华中机床请传 "NCLinkApi" */
    protocol?: DatacollectionProtocol;
}

export const datacollectionApi = {
    async getPaths(deviceId: string): Promise<string[]> {
        const res = await client.get<string[]>("/api/datacollection/paths", {
            params: { deviceId },
        });
        return res.data;
    },

    async list(deviceId: string): Promise<DatacollectionRow[]> {
        const res = await client.get<DatacollectionRow[]>("/api/datacollection", {
            params: { deviceId },
        });
        return res.data;
    },

    async sync(body: {
        deviceId: string;
        items: DatacollectionSyncItem[];
        visiblePaths?: string[];
    }): Promise<{ saved: number }> {
        const res = await client.post<{ saved: number }>(
            "/api/datacollection/sync",
            body,
        );
        return res.data;
    },
};

File: MachineConnectionApi20260423/MachineConnectionApi/MachineConnectionApi/Entities/Datacollection.cs
namespace MachineConnectionApi.Entities;

public class Datacollection
{
    public int Id { get; set; }

    /// <summary>显示名称（对应界面「显示名称」）</summary>
    public string Name { get; set; } = "";

    /// <summary>点位路径</summary>
    public string Path { get; set; } = "";

    /// <summary>节点类型（对应界面「节点类型」，如 Variable）</summary>
    public string Datatype { get; set; } = "";

    /// <summary>保存日期（库字段类型为 date）</summary>
    public DateTime? Datetime { get; set; }

    /// <summary>采集频率（ms）</summary>
    public int CollectionFrequency { get; set; } = 500;

    /// <summary>所属设备 Id，用于多设备隔离</summary>
    public string DeviceId { get; set; } = "";

    /// <summary>Collection route metadata. Legacy NCLinkApi rows also use the Host device driver.</summary>
    public string Protocol { get; set; } = "IndustrialIoT";
}
